### PR TITLE
[source-recharge, source-zendesk-support, source-google-sheets] - Migrate to CDK v4.3.0

### PR DESCRIPTION
**Pull request by pnilan**
## What
- Unpin on CDK 4
- Set python version to "^3.10,<3.12"
- minor version bump

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: docs/integrations/sources/google-sheets.md [hunk 1]
```diff
@@ -166,6 +166,7 @@ Airbyte batches requests to the API in order to efficiently pull data and respec
 
 | Version | Date       | Pull Request                                             | Subject                                                                           |
 |---------|------------|----------------------------------------------------------|-----------------------------------------------------------------------------------|
+| 0.7.0 | 2024-08-02 | [*PR_NUMBER_PLACEHOLDER*](https://github.com/airbytehq/airbyte/pull/*PR_NUMBER_PLACEHOLDER*) | Migrate to CDK v4.3.0 |
```
Comment (girarda) [inline docs/integrations/sources/google-sheets.md:169]:
```suggestion
| 0.7.0 | 2024-08-02 | [42975](https://github.com/airbytehq/airbyte/pull/42975) | Migrate to CDK v4.3.0 |
```

File: docs/integrations/sources/recharge.md [hunk 2]
```diff
@@ -79,6 +79,7 @@ The Recharge connector should gracefully handle Recharge API limitations under n
 
 | Version | Date       | Pull Request                                             | Subject                                                                                                                        |
 |:--------|:-----------| :------------------------------------------------------- |:-------------------------------------------------------------------------------------------------------------------------------|
+| 2.4.0 | 2024-08-02 | [*PR_NUMBER_PLACEHOLDER*](https://github.com/airbytehq/airbyte/pull/*PR_NUMBER_PLACEHOLDER*) | Migrate to CDK v4.3.0 |
```
Comment (girarda) [inline docs/integrations/sources/recharge.md:82]:
```suggestion
| 2.4.0 | 2024-08-02 | [42975](https://github.com/airbytehq/airbyte/pull/42975) | Migrate to CDK v4.3.0 |
```

File: docs/integrations/sources/zendesk-support.md [hunk 3]
```diff
@@ -167,100 +167,101 @@ The Zendesk connector ideally should not run into Zendesk API limitations under
...
-| 0.1.12  | 2022-01-25 | [9785](https://github.com/airbytehq/airbyte/pull/9785)   | Add additional log messages                                                                                                                                                                                                        |
-| 0.1.11  | 2021-12-21 | [8987](https://github.com/airbytehq/airbyte/pull/8987)   | Update connector fields title/description                                                                                                                                                                                          |
-| 0.1.9   | 2021-12-16 | [8616](https://github.com/airbytehq/airbyte/pull/8616)   | Adds Brands, CustomRoles and Schedules streams                                                                                                                                                                                     |
+| 2.7.0 | 2024-08-02 | [*PR_NUMBER_PLACEHOLDER*](https://github.com/airbytehq/airbyte/pull/*PR_NUMBER_PLACEHOLDER*) | Migrate to CDK v4.3.0 |
```
Comment (girarda) [inline docs/integrations/sources/zendesk-support.md:170]:
```suggestion
| 2.7.0 | 2024-08-02 | [42975](https://github.com/airbytehq/airbyte/pull/42975) | Migrate to CDK v4.3.0 |
```

**Review by girarda — APPROVED** [review]

**Comment by pnilan** [comment]
/approve-regression-tests

Simple CDK upgrade – did not run reg tests

**Pull request merged**